File: solution/2485. Find the Pivot Integer/2485_02.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int presum = 0;
        int total = n * (1 + n) / 2;
        for (int x = 1; x <= n; ++x) {
            presum += x;
            if (presum == total - presum + x)
                return x;
        }
        return -1;
    }
};


// [. . . . . {x] . . .}

File: solution/2485. Find the Pivot Integer/2485_02.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        total = n * (1 + n) // 2
        presum = 0
        for x in range(1, n + 1):
            presum += x
            if presum == total - presum + x:
                return x
        return -1
        
